src/quote.rs

```diff
@@ -25,7 +26,21 @@ impl<T: Decode + Into<u64>> Decode for Data<T> {
     }
 }
 
-#[derive(Decode, Debug)]
+impl<T> Encode for Data<T>
+where
+    T: Encode + TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: std::fmt::Debug,
+{
+    fn encode(&self) -> Vec<u8> {
```

Comment:
Prefer implementing `fn encode_to` rather than `fn encode` to avoid one extra heap allocation.